New design implementation

bobbyangelov opened the pull request:
Here is the new design for the website. I tried to follow the ideas shared in https://github.com/syl20bnr/spacemacs/issues/1423. Would be great to get some feedback! 

Live demo: http://boyanangelov.com/spacemacs/
Here are some screens:

![screen shot 2015-11-05 at 16 46 37](https://cloud.githubusercontent.com/assets/4747808/10973242/d9f6fd76-83dc-11e5-89eb-44a97042df4e.png)
![screen shot 2015-11-12 at 09 54 52](https://cloud.githubusercontent.com/assets/4747808/11114244/8143316c-8923-11e5-915c-70ba7d06e9e7.png)
![screen shot 2015-11-12 at 09 55 00](https://cloud.githubusercontent.com/assets/4747808/11114242/8054fa7e-8923-11e5-85e8-5bb870ab2dc0.png)

d12frosted commented:
WOW

cpaulik commented:
Why share screens when the site is online at http://boyanangelov.com/spacemacs/ ?

Anyway I think it looks great although very familiar. But maybe I'm just a little bit annoyed that all the sites tend to look so similar nowadays.

TheBB commented:
That's sexy.

What I would like to see on this page is the main documentation org files as well as the layer READMEs rendered in something like readthedocs format. Maybe using something like [ReadTheOrg](https://github.com/fniessen/org-html-themes#readtheorg)? How automatic can we make this?

bobbyangelov commented:
@d12frosted Thanks, its based on a theme (see the attributions), honestly I didn't do that much :)
@cpaulik Completely agree... Even if those designs are very minimal and work great on mobile, we should try to do something more inventive in the future, I think that would be the spacemacs way :)

cpaulik commented:
Could we harmonize the font/spelling of this:

![Spacemacs logo](https://raw.githubusercontent.com/syl20bnr/spacemacs/master/doc/img/title2.png)
 
from the README with the website?

I have no opinion on which to choose but I think they should be the same.

A fixed-width font for the logo would also be fitting in my opinion.

robbyoconnor commented:
I LOVE THIS!

robbyoconnor commented:
Some comments though -- maybe link to both `master` and `develop` docs :D

TheBB commented:
Following up on the documentation thing. I was able to convert `doc/DOCUMENATION.org` into a format that enabled it to be exported to HTML. Then applying the theme was not difficult.

![result](http://i.imgur.com/V2pYABp.png)

Mostly the difficulty is supporting both github and org-mode style links at the same time. I wasn't able to export until I had converted all links to org style. The disadvantage with this is that the doc files are no longer useful on github. **IF** we can get all of them on spacemacs.org, I think that's not necessarily a problem. We can make that the official way of accessing the docs online, and reading it in Emacs is still not a problem. The README.md file would stay and the links there could be updated to point to spacemacs.org. This also would let us scrap `toc-org` which we've had problems with in the past (although it's been fine for a while now).

Thoughts?

You can see my work in my [readthedocs](https://github.com/TheBB/spacemacs/tree/readthedocs) branch. It basically consists of:

* removing the auto-generated table of contents
* changing all the links
* adding the style headers

It is of course also possible to write a `spacemacs/export-documentation` function that does all this automatically, to generate files that can then be pushed into the `gh-pages` branch. Presumably we would have to write such a function eventually anyway (to deal with all the different org files we have simultaneously), but this way strikes me as the path of least (but not little) resistance and maximal awesomeness.

syl20bnr commented:
@bobbyangelov Thank you so much! My feedback in random order:

- I like the layout and the straight to the point presentation.
- I don't like so much is the gradient and the choice of colors which make the logo not very visible.
- I like the idea of keeping the theme of space in the colors, so something more dark as a background that place the Emacs planet in context would be wonderful. I also like the "pun" with _Space_macs.

Anyway this iteration is already a _huge_ improvement over the current broken site and we are free to tune the theme later. But if you (everybody) have some alternate propositions before we upload it, don't hesitate :-)

@cpaulik I agree, would be great to have the fonts match.

I add @nashamri in the loop who is the author of the logo.

@TheBB Excellent ! I'm for more flexibility so being stuck on the GitHub platform for the doc is not something that I'm fond of. Moving the doc to a website is an appealing option.

syl20bnr commented:
Also would it be possible to try a gradient from top to bottom instead of from left to right ?

Crazy idea would be to go from space to the land of the Emacs planet, with somewhere the orbiting Evil flagship.

The horizon line could mark the start of the description of the feature.

bobbyangelov commented:
@syl20bnr I have an idea about the gradient background :art:! I will work on that, the logo and a few minor things and then submit for one more round of feedback.

@TheBB that looks very cool! I'm not sure if I can help you, but tell me if there's something I can do.

thank you @robbyoconnor :-)

TheBB commented:
@bobbyangelov It's fine. I will keep working on this and I might make a PR against your branch eventually. You don't have to wait for me though, if you finish up then I can make another PR against `gh-pages` later on.

bobbyangelov commented:
Cool :+1:

stormpat commented:
@bobbyangelov Very nice work!

nashamri commented:
That's great @bobbyangelov.

I would suggest using something like this for the gradient:
![sitesuggestion](https://cloud.githubusercontent.com/assets/1207107/10855768/5ae79f22-7f3b-11e5-98f8-4d5988dff98b.png)

If you guys like this, I can elaborate on this idea and send it to @bobbyangelov.

Also, I noticed that the shadows are removed from the logo, I don't know if that's intentional or not, but the shadows are meant to be the letter "E" from Emacs :smile: so they are kinda important.

And yeah I'm with @cpaulik for a new font. The font that I put for the logo was almost a place holder :smile: 

Another note about the screenshot, please use this one 
https://github.com/syl20bnr/spacemacs/commit/668d20700c5da8d73612b012910df738fa6ef73b
:smile:

TheBB commented:
@bobbyangelov

I can't figure out how to make a PR against your fork, but can you grab this commit from me? https://github.com/syl20bnr/spacemacs/commit/91b5e1f860594b249d1ccee92d9ef8b9ee73a2e3

It includes all the org-files in the `doc` and `layers` directories in readthedocs-like format. There are some rough edges but it should be enough to get you started on figuring out how to integrate them into the rest of the site. :-)

For the rest of you, if you want to have a look:

http://thebb.github.io/spacemacs/doc/doc/DOCUMENTATION.html

replace DOCUMENTATION with CONVENTIONS or VIMUSERS or whatever.

bobbyangelov commented:
@nashamri Love the new background, we should go for it! About the shadows - accidental editing on my part when creating a font logo :neutral_face: Do you have ideas for fonts? 

@TheBB Great, will merge that :-)

nashamri commented:
@bobbyangelov Is there a repo for the site? I'm thinking what's the best way to collab on this.

bobbyangelov commented:
@nashamri I added you as a collaborator on my fork.

syl20bnr commented:
I like this concept https://github.com/syl20bnr/spacemacs/pull/3604#issuecomment-152625314 :heart: 

@TheBB I wonder if we could already merge your changes to the documentation, it will be in develop branch only so it should not bother too much if the links are broken on GitHub while we prepare the website. It would be simpler then for the people working on the website to just take the develop branch as a reference. I'm late on the planed feature for 0.105.0 but I think we could release it combined with the new website and move some 105 items to 106.

TheBB commented:
Yeah, we could do that. I already mentally signed off for tonight but I can prepare a PR tomorrow.

bobbyangelov commented:
Hi everyone! I pushed the latest changes. I added the links to the documentation from @TheBB , and the background from @nashamri . What do you guys think?

TheBB commented:
I left these comments on gitter, copying here for posterity.

* I like it!
* I feel `/layers/LAYERS.html` deserves a link, and possibly a better styling. (I don't know much about styling.)
* If the quick start should get such a prominent link we should revise its contents. I can do that in #3613.

nashamri commented:
Super! I love it :+1:

TheBB commented:
Did any of the images from the export make it? Can't see any.

a13ph commented:
@bobbyangelov 
I like it - it's pretty and not too crowded. 


However: 
* Try looking at it on smaller window sizes (emacs and tiling vms are not so rare to live together, and it's simple test for fluidness for mobile browsers too). 
* I screenshot not being displayed cause of fairly eager script blocker. Why do you even need script from ajax.google.com for a screenshot?
* screenshot is too small and unreadable, even if I try to zoom. Showing off text editor without showing off how it displays text is questionable
* git clone command box is scrollbarred on small window size, despite there being plenty of width to use.
* I'm unsure about metaphors, especially "Display" for discoverability. Sure, graphics and all, but display feels way too generalized. 
* "Building blocks" at least have this common phrase going for them, even though it's not quite = consistency. 
* "Fork" is not exactly crowd-*configuration* symbol, but it's pretty okay with how iconic this symbol is for github collaboration of many flavors and how this (can I even call it *our* at this point?) project is mostly about configuration of configurations for text environment  :potable_water: on top of a Lisp Interpreter
* "Keyboard"  does not imply mnemonics, but It gives mental image of mnemonic keybinding layout in just a word and a symbol. 
* which leads to the point of making symbol, big text and small text being coherent in each group together
* and probably much better if there would be consistency in how big text + small text read together - is that a sentence or being a pair of short slogans or whatever you'd call those phrases.
* Landing page needs as little text and scrolling as possible, even for such product.
* User focus should be guided in a most likely path, not flooded with links to various documentations.
* No license of site or Spacemacs or Emacs anywhere on the site :registered: 
* badges are distracting for such a high contrast background
* contrast between dark and light areas should probably be reduced, page down in darkness may be pretty blinding. It's not even light backgrounds which hurt, but high contrast. 
* That doesn't mean text to background contrast should be low. "Recommend" in Slant badge is guilty of this AND of enormous contrast compared to other, more important elements - even, no, especially if spreading out word about Spacemacs is one of highest priorities.


Some of those are pretty opinionated, but I tried my best to find ways to improve that page.

bobbyangelov commented:
I implemented a few adjustments based on feedback from @a13ph and @TheBB . @TheBB , yes, there is something wrong with the images in the documentation :confused:

TheBB commented:
I'll have a look-see on those images.

TheBB commented:
Oh, they literally didn't make it, as in they're not there. That should be fixable.

cpaulik commented:
Would it be over the top to theme the documentation like the spacemacs theme?

bobbyangelov commented:
@cpaulik I'll see what I can do :art:

TheBB commented:
I just pushed an update with newer and nicer documentation (with images this time). I had to fix some bugs in the publishing code, will send that to Spacemacs in a bit.

The documentation is using the [readtheorg theme](https://github.com/fniessen/org-html-themes#readtheorg) which is defined by [these](https://github.com/fniessen/org-html-themes/blob/master/setup/theme-readtheorg.setup) linked files. Someone who knows CSS (i.e. @bobbyangelov and not me) could probably link them to a local file with other colours.

In fact we should probably serve those files from our own server in any case.

One note though, I recommend not to get *too* carried away with the doc colours. At the very least keep it a light theme.

cpaulik commented:
@TheBB Yes we do some light `spacemacs light` touches on the doc theme. So that is does not look too generic.

Also when exporting the org files you should set a matching theme since org html export hard codes the currently set theme colors into the HTML.

cpaulik commented:
I really like the metallic version of the spacemacs logo.

bobbyangelov commented:
@cpaulik thanks to @nashamri for the great design work :smiley:

nashamri commented:
And I just changed the metallic logo @cpaulik lol

cpaulik commented:
Still looks good @nashamri :smile:

nashamri commented:
Thanks guys, I'm giving the documentation theme a shot.

syl20bnr commented:
Amazing job guys :purple_heart: 

I find the front-page a little bit overcrowded, can we try a design:
- without the badges
- without the shell command
- replacing `QUICK START` by `QUICK INSTALL` that unveiled the shell command
- move the quick start link to the links line

bobbyangelov commented:
@syl20bnr Done! I also added the second layer link ( @TheBB ).

TheBB commented:
Great, I only have one more comment until this is good to go as far as I am concerned.

I agree with removing the badges but I still think a link to gitter is nice. Maybe it can go in the top right? Unfortunately font-awesome doesn't have a [gitter icon](https://github.com/FortAwesome/Font-Awesome/issues/6535) but I don't think that's a deal breaker.

cpaulik commented:
The source code blocks should still be fixed. 
There are two ways to do this

- @TheBB exports the org files using a light theme so that the source code is highlighted in dark colors
- @nashamri modifies the theme so that source code blocks have a dark background.

Otherwise I :heart: it.

TheBB commented:
Okay, @cpaulik, I just now realized what you were talking about. I'm a little taken aback that org would do this. Seems a little counter-intuitive. But okay, I guess it makes sense to export with spacemacs-light?

cpaulik commented:
Yeah it is a little strange that this is hardcoded and not solved with a CSS file.

Spacemacs light should look good.

TheBB commented:
Yeah, it looks better.

I couldn't get an absolute link to work. I tried moving @nashamri's `readtheorg.css` to `/css/readtheorg.css` since it will be accessed from many different places, but no deal. What's the best way to do this on github pages?

Awaiting the resolution of :point_up: I have not pushed a doc update to this branch yet.

syl20bnr commented:
@bobbyangelov GREAT, this is just beautiful :purple_heart: 

One last query and I think I'm done with it :smiley: 
Can we try to replace the beige background to a purple one that match the end of the gradient ? I would like to have to feeling that we leave the sky smoothly. The font color could be switch to white.

syl20bnr commented:
Also for the overall symmetry of the page the two buttons `Download` and `Quick Install` should have the same size. We could drop the `Quick` also.

bobbyangelov commented:
@syl20bnr What do you think now? Should I change the beige color as well? :-)

syl20bnr commented:
I like it but I had in mind to use the last colour of the gradient in order to avoid hard separation between the front page and the screenshot page.

Keeping beige for the third page is OK with me.

Let see what the other think about this change.

cpaulik commented:
I think the idea of having the gradient go to the background color of the next section is neat. But there is a problem with the background image used.

The very left bottom pixel seems to have the color `#565482` while the very right bottom pixel is `#59598F`. So this will never look as smooth as we want it.

cpaulik commented:
Also there is a few pixels dark area on the left border of the backgroud image which makes it look weird when the transition should be smooth.

See 
![image](https://cloud.githubusercontent.com/assets/380927/10950561/8248c8c4-8338-11e5-9538-c30e9ffb1c0e.png)

bobbyangelov commented:
@nashamri Can you help out with the backgrounds?

nashamri commented:
Done @bobbyangelov, I found that the old gradient has some problems when viewing the site on multiple screen sizes. The quick fix was to raise the base gradient color a bit. Now it looks okay on mobile to 1080p screens. Don't know how it'll look on HiDPI screens though.

TheBB commented:
I honestly preferred the previous version. (With a sharp edge.) Now that area between looks so empty. Maybe we could move the screenshot up a bit to compensate.

How is the doc CSS stuff? I don't mean to badger you about it. :-P

nashamri commented:
@TheBB no progress on the CSS stuff yet lol.
Probably the old hard edge is better, I'm kinda leaning towards the old look. Now that we saw them both, which one do you guys think is better?

syl20bnr commented:
Thank you for this new iteration, I also prefer the previous hard edge.

syl20bnr commented:
Actually the more I watch it the more I like it but the screenshot is definitely out of place, it is to close to the third page edge.

syl20bnr commented:
I'll look at it later again on a different resolution.

cpaulik commented:
The paddings should be adjusted slightly. There is no padding below the screenshot so it is too close to the edge compared to e.g. the elements in the beige section.

I have not strong preference for the hard edge vs. fading look.

Another matter is the text. I think we should use the new copywriting introduced in https://github.com/syl20bnr/spacemacs/pull/3332 also on the website.

syl20bnr commented:
![capture d ecran 2015-11-04 a 20 58 22](https://github-cloud.s3.amazonaws.com/assets%2F1243537%2F10957899%2Fe5ecd4bc-8336-11e5-9298-66d8aeb28965.png)

It is really nice. I guess now it is a matter of tweaking.
- The `Spacemacs_`is maybe to close from the logo, we could have the same space between the logo and the buttons,  maybe the logo need more room to distinguish itself even more. 
- The stars in the purple part should be removed, stars are OK in the sky but not on the planet :-D
- I don't know if it is possible but can we bend a bit the gradient to make it radial ? (but very slightly, just to suggest that this is a planet).
- I would love some parallax effect on the planet and text when scrolling down coupled with a fade out of all the text elements, making it really like we are landing.
- Iooks like we still need a hard edge to separate the beautiful first page with the screenshot page, maybe a darker purple ? I like the feature being in beige.

syl20bnr commented:
Also we need to find the best place for the transition between space and the planet, first it was low so the logo was high in the sky, now it is much higher to the logo is closer to the planet. Not sure were to place it to get the best effect, especially if we achieve to have a radial gradient.

syl20bnr commented:
In the case of parallax, it would be awesome to be able to animate the gradient to go up faster than the space stars plan.

nashamri commented:
@syl20bnr I didn't think the gradient meant to be a planet surface :smile:  but I love the idea and having a curved horizon would be cool if I can do it right! I'm on it!!

nashamri commented:
There's a small problem with it now when the page is scaled up and down (It could be a cool unintentional effect :smile:). I don't know what's the best way to tackle this issue though.

stshine commented:
Is it possible to add an article section for publishing news, tips & tricks, reviews and so on?

bobbyangelov commented:
@stshine Yeah, sure. But then would be great to make a dynamic website (i.e. with jekyll). Maybe for a second version?

stshine commented:
@bobbyangelov Yes, since that would not be some trival work in both design and coding it would be better to leave it later.

syl20bnr commented:
@nashamri :purple_heart: :purple_heart: :purple_heart: This is beautiful!

I hope we can make it responsive, it is so beautiful on a desktop in full screen.

gregpardo commented:
![Mugatu](http://forum.malazanempire.com/uploads/monthly_09_2014/post-3948-0-02129200-1410304547.jpg)

bobbyangelov commented:
I added some changes based on the feedback. @nashamri the planet looks so awesome! I think we should change the background of the beige section as well.

nashamri commented:
@syl20bnr thanks :smile: 
@gregpardo lol thanks 
@bobbyangelov great job now it looks cool on all sizes.

cpaulik commented:
Now the only thing that is missing is the music from https://youtu.be/Y15NnGZIBuM?t=149 :smile:

nashamri commented:
@cpaulik loooooooool I was exactly thinking about 2001 when I finished the background!

gregpardo commented:
I know it's just alpha but wanted to comment that the favicon is not hires. It looks pixelated on retina mackbook.

syl20bnr commented:
> I think we should change the background of the beige section as well. Maybe add a similar horizon, so that it looks below the planet?

We can experiment with different colors, the idea of the planet is good but I would see it more for the footer and keep the 3rd page in the planet but the lack of separation between the 2nd and 3rd may be problematic.

syl20bnr commented:
I find the logo too small on mobile:

![img_0250](https://cloud.githubusercontent.com/assets/1243537/10978305/2e2e9bc4-83c4-11e5-8a3d-29e92b9d365b.PNG)

syl20bnr commented:
@bobbyangelov BTW nice tweaks, I like the new layout and the planet horizon is now fixed on mobile :+1:

nashamri commented:
I just removed the scroll bar lol. It might be a stupid move, waiting for your comments.

TheBB commented:
I can't scroll using my touchpad. Not sure why that is. Guessing it's related.

cpaulik commented:
I can not scroll anymore using any method

syl20bnr commented:
Same here, arrows work but other means don't work anymore.

nashamri commented:
It was working fine with my laptop when I two-finger scroll. When I tried my other laptop, it didn't work. So, this was a FA!L :smile:

bobbyangelov commented:
I increased the logo size for smartphones and optimised the favicon. Do you guys have more suggestions or can we deploy this? :stuck_out_tongue_winking_eye:

TheBB commented:
We have some work to do with the documentation still, but I'm good with deploying this now and fixing that up in a later PR. So long as it happens before 0.105.

syl20bnr commented:
We should have a different backgrounds (front page and footer) with a ratio close to main mobile devices screen ratio, right now the stars are wrong on mobile.

nashamri commented:
@TheBB regarding the code blocks issue, I think I've found a *hack* which is not an optimal solution, but worth a try, I'll push a little test soon.

bobbyangelov commented:
I changed the background color of the third section. This is the complementary color to the purple above, so I think it looks nice? Also made the footer smaller in height and removed its background. Here's a screenshot:

![screenshot 2015-11-07 18 59 25](https://cloud.githubusercontent.com/assets/4747808/11016428/bd5e9fb0-8581-11e5-8318-c6bc11a675cb.png)

nashamri commented:
I like the old one more :worried:

nashamri commented:
I'm sorry for not posting screenshots instead of actually committing the changes (879a1e7), but I want as many of you to test this as I'm not sure it'll work on all platforms/browsers.

TheBB commented:
Not exactly what I had in mind. I can have a look tomorrow if someone can answer me this: what's the best way to get an absolute link to the root of the repo that will work both on spacemacs.org and on github.io?

TheBB commented:
:point_up: @bobbyangelov or @naseer? As for the color issue I'm kind of leaning towards the all-blue design.

nashamri commented:
@TheBB there aren't that much that can be done on the CSS side of things, I'm afraid.

TheBB commented:
Well, I'll give it a shot. If it doesn't work we just have to export with a specific theme set.

StreakyCobra commented:
[Maniac speaking here] 3 over the 4 icons are not vertically centred inside their circles :smiling_imp:

nixmaniack commented:
I'm finding the whole blue theme nice. 

Also, there's quirky animation when you click Install button. It just jumps weirdly.

syl20bnr commented:
I prefer the blue one as well even if the old beige was a reminder of the light theme of Spacemacs. I cannot see the blue background on my mobile.

- The smaller footer is fine but I liked the idea of puting the end of the planet in the footer and having space and stars again in it.

- Also maybe a bug in screenshot https://github.com/syl20bnr/spacemacs/pull/3604#issuecomment-154089332 the link in the top right corner are not pushed all the way for the right.

- the menu on the top right corner for mobile need some work.

The logo size is now good on mobile.

I don't know how much polish we want before pushing but it worth fixing as much as we can before going live, we don't have much pressure of a milestone :-)

nashamri commented:
I think the padding between sections needs a bit of work.

syl20bnr commented:
That's nice but my initial idea was really to end the planet for the footer not the 3rd page (so the footer would be in space with stars like the first page).

For the background color of the feature page I'm not sure about my preference, the initial beige is warm and the new blueish is cold, depends on what we want. Since I hesitate maybe the beige was better.

![capture d ecran 2015-11-08 a 12 49 08](https://cloud.githubusercontent.com/assets/1243537/11021656/7b7a9428-8617-11e5-9e23-51ff795a0d3d.png)

TheBB commented:
I pushed a documentation update. It now uses the spacemacs light theme, fully through CSS and not hardcoded. The corresponding PR to Spacemas proper is #3725. I'd appreciate if someone familiar with org could look at the issue there. It works, it's just not super pretty. :-)

TheBB commented:
As for the design, I actually quite like it this way.

TheBB commented:
![Screenshot](http://i.imgur.com/BOi4eQc.png)

nashamri commented:
@syl20bnr yeah I guess having the end of the plant in the footer section makes sense. :smile: 
I'll do that tomorrow and try the two colors and see which is better.

@TheBB Great job :+1: that looks so pretty :+1:

naseer commented:
Loving the new design :+1:  :)

syl20bnr commented:
@TheBB Nice work! One remark: the code blocks are not easy to distinguish because of the same background color.

nashamri commented:
I could just make the page background white?

On Mon, Nov 9, 2015 at 7:17 PM, Sylvain Benner <notifications@github.com>
wrote:

> @TheBB <https://github.com/TheBB> Nice work! One remark: the code blocks
> are not easy to distinguish because of the same background color.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/syl20bnr/spacemacs/pull/3604#issuecomment-155161675>.
>

TheBB commented:
Yes, that would be nice.

Maybe I'm being blind but I can't see on the current page the same colours as the screenshot I posted. On the website the code blocks appear to have nicely distinct colours. Viewing them on the same monitor, too. Oh well.

nashamri commented:
I think there's something quirky happening with the css, sometimes I see a white background and sometimes it's beige. I'll just make it white as it was originally :smile:

bobbyangelov commented:
@nashamri oh no, I'm on it :neckbeard:

nashamri commented:
@bobbyangelov that looks good now :+1:

nashamri commented:
![proto](https://cloud.githubusercontent.com/assets/1207107/11100532/b35cc016-88a9-11e5-84bc-b7acdd75e69a.png)

Shall we use this? or stick with the current design?

bobbyangelov commented:
@nashamri looks awesome to me! :+1:

syl20bnr commented:
I :purple_heart: this !

syl20bnr commented:
(I miss the little dots to separate the links in the top right and bottom right corners :angel:)

joehillen commented:
Looks awesome! 

Can we add more screenshots though? It would be nice to show more themes and features (which-key for example). Maybe the screenshots could be an image slider with buttons.

syl20bnr commented:
I propose to drop the sub titles _Key bindings_, _Total control_, _No decision fatigue_ and _Open Source power_.
They don't bring useful information and seems artificial.

syl20bnr commented:
> Maybe the screenshots could be an image slider with buttons.

:purple_heart:

nashamri commented:
![proto2](https://cloud.githubusercontent.com/assets/1207107/11102138/ba057f8a-88b2-11e5-9c4c-e272de243433.png)

+1 for an image slider with more screenshots!

syl20bnr commented:
@nashamri oh not the entire page :smile: 

I was for removing the subtitles only:
- Key bindings
- Total control
- No decision fatigue
- Open Source power

nashamri commented:
@syl20bnr looool okay

nashamri commented:
![proto3](https://cloud.githubusercontent.com/assets/1207107/11102980/bae8ae72-88b7-11e5-90f0-57ab686762ee.png)

This is just an idea not a real implementation.

StreakyCobra commented:
@nashamri I like it alot this way also :+1:

syl20bnr commented:
_O-M-G_ so **beautiful** :purple_heart: :purple_heart: :purple_heart:

syl20bnr commented:
The icon for *Discoverability* could be some kind a `?` or something that remind `help` (i.e. F1). Not sure what symbol to use though.

StreakyCobra commented:
(except the icons not centred on their circles :scream: )

joehillen commented:
@StreakyCobra you're asking for the impossible ;)

syl20bnr commented:
For the _Crowd-configured_ I would like to put people instead of a fork (which is good but I prefer to focus on people than the collaboration model).

joehillen commented:
> (except the icons not centred on their circles :scream: )

Really the only way to consistently avoid that issue is to just not have the circles. 

:fire: CSS :fire:

nashamri commented:
@syl20bnr what about a magnifier glass for Discoverability? similar to a search icon.

syl20bnr commented:
:+1:

StreakyCobra commented:
> Really the only way to consistently avoid that issue is to just not have the circles.

Or to use image instead ^^

nixmaniack commented:
That's beautiful!! Amazing work by both @bobbyangelov and @nashamri.

joehillen commented:
There should probably be a link to the stickers: https://www.stickermule.com/marketplace/6839-spacemacs-logo

@syl20bnr Any plans to sell some t-shirts? :smile:

TheBB commented:
This is probably me being picky but... two of the 'pillars' are nouns and the other two are adjectives (I read mnemonic as an adjective since I think that's how it's used here). I think it would flow better if all were the same word class, probably adjectives.

So I propose 'discoverable' and 'consistent' in place of 'discoverability' and 'consistency'.

nashamri commented:
@TheBB :+1:

nashamri commented:
Just pushed an image slider, let me know if there's something wrong with it.

bobbyangelov commented:
@nashamri man that slider looks so cool! :green_heart:

cpaulik commented:
Great work. 

Shouldn't the third menu point be `layer list` instead of `layers list`?

nashamri commented:
@StreakyCobra how bad are they off the center? Can you send a screenshot. They look okay on my end but I'm thinking it might be a rendering issue on your end. If that's the case then I'll use images instead.

StreakyCobra commented:
@nashamri Vertical centering
![2015-11-12-102220_3360x1050_scrot](https://cloud.githubusercontent.com/assets/1587877/11114878/1b0cfe6e-8928-11e5-9f64-5e7d439875d0.png)

joehillen commented:
I'm going to advocate for the "less is more" design pattern and suggest just removing the circles. 

They don't add anything, and they will never be perfect in all browsers on all platforms.

StreakyCobra commented:
Yeah and zooming/unzooming the page is even worse :joy: At 72% they are kind of centered. I like the circle, it's just maybe better to just create images from them, they will scale and distribute better.

nashamri commented:
Did a tiny tweak for the centering issue, looks okay to me now. what do you guys think now?

StreakyCobra commented:
@nashamri Yes it's better, still shifted a few pixels, but it's acceptable now. On the other side, zooming/unzomming still destroy it totally. Just pointing it out :relaxed: 

I was really liking your version with other screenshots in perspective on the sides, aren't you gonna to try this one?

nashamri commented:
@StreakyCobra zooming/unzooming did destroy it under firefox, not under chrome though :smile: 
Regarding that perspective slider, I might give that a shot, it could be difficult since we can't get a circle centred correctly nowadays lol #NoStandardsAreTheNewStandard

StreakyCobra commented:
@nashamri 

> zooming/unzooming did destroy it under firefox, not under chrome though

I see your troll here :wink:

> Regarding that perspective slider, I might give that a shot, it could be difficult since we can't get a circle centred correctly nowadays lol #NoStandardsAreTheNewStandard 

Go ahead, I really want to see it live!

nashamri commented:
@StreakyCobra nah I love firefox lol. Hope it's better now.

gregpardo commented:
The icons look good to me now. Although the 'consistent' icon appears a little off to the left but perhaps might be an optical illusion.

CestDiego commented:
we need this naaauuu

naseer commented:
:+1:  - could do more tweaking later when needed? It's already looking great!

TheBB commented:
I think it's mergable, too.

syl20bnr commented:
Last thing before making it live is to improve the text in the feature section, unfortunately I don't have time this evening to propose something and it's hard to explain what I would see in this section.

syl20bnr commented:
Oh I forgot, the docs are not readable on mobile, the left pane takes 80% of the screen.

nashamri commented:
@syl20bnr Now the situation should be better for mobile. For the feature text, are talking about the "wording" or the actual font it self? I can play with other fonts if that's the case and see what you guys like.

syl20bnr commented:
I was talking about the wording, the font is fine but if you have other propositions then I want to see them :-)

I would like to publish the new website at the same time of the release of 0.105, I will start to work on it in the next few days.

nashamri commented:
Hmmm, we need someone who's good with marketing :smile: as those "features" sentences are coming straight from the documentation. I especially don't like the "Discoverable" section:
```
Spacemacs comes with a dedicated spacemacs-mode. Its goal is to give useful feedback and simplify maintenance tasks
```
I don't think this is useful info for newcomers.

bobbyangelov commented:
Nice :) Already looking forward towards working on the dynamic version. :neckbeard:

syl20bnr commented:
The docs are now readable on mobile but the navigation sidebar is gone. Is it possible to add the header like in Ansible documentation ?  http://docs.ansible.com/ansible/intro_inventory.html

syl20bnr commented:
On mobile the link line takes too much place (3 lines), I propose to move them in the toggled menu.

gregpardo commented:
Small critique here. Can we change the body background-color to the dark blue space color? I'm on a mac with chrome and when you scroll past the top or bottom of the page you see the white exposed and it looks odd.

nashamri commented:
@syl20bnr this 27d9d13 almost drove me nuts lol :smile: . I wanted to make the navigation sidebar toggelable without adding javascript and without adding too much markup to the html file. I wanted the heavy lifting to be done by css. I think I kind of got it working now. only 2 lines of markup, and the rest is css. This should make it easy for us to update all the layers docs.

I've only changed the FAQ page to get your feedback before we generalize it to the other pages.

nashamri commented:
@gregpardo d461e94 should fix it. I don't have a mac to test it though, let me know if it's resolved.

gregpardo commented:
@nashamri Body bg fixed here.

syl20bnr commented:
@nashamri sidebar is working (tested on mobile safari) :+1:

nashamri commented:
@syl20bnr cool, I can update the reset of the pages except the layers docs. The layers docs need a bit of effort as they are now generated by `org-publish` and I need to figure how to inject the needed markup there.

nashamri commented:
@travisbhartwell well done :clap: this should solve the last remaining issue with the documentation. Could you PR it against the syl20bnr develop branch, please :smile:?

Just letting @TheBB be in the loop.

TheBB commented:
Looks good to me. So long as it's all wiped away after publishing I don't mind if it's a bit hacky.

TheBB commented:
I pushed a new doc version with @travisbhartwell's changes.

TheBB commented:
I found four more layer docs without the magic cookie in the header. They're fixed now in develop and will be good next time someone pushes a new documentation update to this branch. They are sql, smex, spell-checking and spotify.

nashamri commented:
Great! I think this PR is Cherry-pickable now :smile:

syl20bnr commented:
** Wait for the 0.105 release please ** :-)

Also we need to edit those feature paragraphs, even if I wrote them in the first place, I don't like them :-D
Another thing, what are those attributions in the footer ?

nashamri commented:
I'm so excited for 0.105 :smile: 
Regarding the attributions, I'm not really sure, @bobbyangelov might know.

bobbyangelov commented:
The attributions are for the theme on which the site is based on, http://tympanus.net/Freebies/Landio/ :) I think those guys deserve credit.

syl20bnr commented:
Ok for credits but currently it is weird, it is like they did the whole site which given the amount of work you guys put in it, is a bit too much.

nashamri commented:
Maybe we could have a `credits` link that points to a separate page and there we can explain the credits with more elaboration.

StreakyCobra commented:
Or simply something simple like

    Attributions:
       Spacemacs — Sylvain Benner & Contributors
       Website   — <your names>
       Template  — Peter Finlan and Taty Grassini

bobbyangelov commented:
@StreakyCobra :+1:

syl20bnr commented:
I'm OK with @StreakyCobra proposal.

gregpardo commented:
+1 his proposal is clear

bobbyangelov commented:
Ok, I will do that then.

synic commented:
It'd be nice if the documentation had a search.  "NeoTree" docs are under "Commands", but you'd never know that unless you expanded it.

StreakyCobra commented:
@synic The whole documentation is one single page, so `Ctrl+F` or similar works. User may not know this though, so a search input can be nice.

bobbyangelov commented:
![screenshot 2015-11-23 21 31 20](https://cloud.githubusercontent.com/assets/4747808/11348916/95656408-9229-11e5-90cf-fc83627e3e01.png)

Is it good so?

StreakyCobra commented:
Use either "and" or "&" for all lines :-) And it would look better if names
start on a  vertical alignment.

TheBB commented:
If there's a choice, please use 'and'.

CestDiego commented:
Can we have this as well somewhere? https://sidecar.gitter.im/ :P

bobbyangelov commented:
@CestDiego wow, that's pretty awesome. What do you guys think?

nashamri commented:
That seems pretty cool :+1:

nashamri commented:
Should we remove the "gitter" button at the top since we have the side chat now?

StreakyCobra commented:
For me we can keep it… or remove it… it doesn't really matter :-p

Otherwise, I can't close the sidecar pane once opened, the close button doesn't seem to be accessible.

nashamri commented:
@StreakyCobra I've experienced some glitches with the close (X) button. But usually it takes a second or two then I can close the side pane.

StreakyCobra commented:
Ok, it's a matter of scrolling the page, not the chat. It seems that the top menu is above the chat, preventing to access the X button.

nashamri commented:
Seems like a bug in their code. Maybe it could be a css issue? z-index?

StreakyCobra commented:
> z-index

It's what I was thinking about, but I'm *really* not a CSS/web guy :-) If this is the problem maybe you can override it with an `!important` ?

bobbyangelov commented:
@nashamri seems fixed :+1:  :grinning:

syl20bnr commented:
This chat sidebar is awesome ! :purple_heart: 

Funny, the maybe new website for GNU Emacs is somewhat similar for our website (same kind of colors but more vivid and feature circles :-D) http://nicolas-petton.fr/ressources/emacs-website/

nashamri commented:
Wow what a coincidence! or not? lol

d12frosted commented:
Wow, nice improvements! Also I like documentation pages. It's a pleasure to read them online. Really good work. 

So far I don't like only one thing - how jumpy content is when you press 'install' button (to show or hide cmd).

d12frosted commented:
Also, sometimes I don't see all round items (don't know how to call them) like in following screenshot:

![](https://www.dropbox.com/s/v8sokhgupuijilw/Screenshot%202015-12-01%2009.23.56.png?dl=0)

It happens when I scroll softly. Usually second and forth are not seen for a while.

CestDiego commented:
can we have http://spacemacs.org serve this now? :(

StreakyCobra commented:
@CestDiego https://github.com/syl20bnr/spacemacs/pull/3604#issuecomment-158940036

nashamri commented:
Bugs in the docs:
* Image not shown in jabber, gnu, lua and eyebrows layers (maybe because they are gifs? @TheBB).

* weird rendering in the emoji layer (http://boyanangelov.com/spacemacs/layers/+fun/emoji/README.html).

* The following layers have an issue loading the css file (smex, spell-checking, spotify, syntax-checking, themes-megapack, theming, typography, mu4e, sql and spacemacs-layouts). I think the issue is in this line https://github.com/syl20bnr/spacemacs/blame/develop/layers/%2Bwindow-management/spacemacs-layouts/README.org#L2 it should look like this https://github.com/syl20bnr/spacemacs/blame/develop/layers/%2Bwindow-management/eyebrowse/README.org#L2

TheBB commented:
Thanks. I can fix those. May have to go hard on the emojis though.

nashamri commented:
Oh those emojis! lol 
Maybe just create a png of them to make it consistent with the rest of the layers.

TheBB commented:
Updated the docs!

syl20bnr commented:
It's in the box, great work guys ! :purple_heart:

bobbyangelov commented:
@syl20bnr great! I am open to start working on a dynamic version in the near future :)

syl20bnr commented:
*IMPORTANT* @bobbyangelov do not delete your PR branch, I will try to keep the history in spacemacs.org repo, thanks!

bobbyangelov commented:
@syl20bnr Sure, I'll keep it. Great that you made a new repo, everything is much more organised.
